spark/src/main/scala/org/apache/spark/sql/delta/DeltaErrors.scala

```diff
@@ -2461,6 +2461,11 @@ trait DeltaErrorsBase
     )
   }
 
+  def identityColumnAlterNonIdentityColumnError(): Throwable = {
+    new AnalysisException(
+      "ALTER TABLE ALTER COLUMN SYNC IDENTITY cannot be called on non IDENTITY columns")
```

Comment:
You cannot use `new AnalysisException`. This doesn't compile against Spark Master.

Please use `new DeltaAnalysisException` and use a proper error class

Reply:
Updated the Exception class.